agent_api/src/main/java/dev/aikido/agent_api/storage/ServiceConfiguration.java

```diff
@@ -87,68 +85,39 @@ public boolean isIpBypassed(String ip) {
      * Check if the IP is blocked (e.g. Geo IP Restrictions)
      */
     public BlockedResult isIpBlocked(String ip) {
+        BlockedResult blockedResult = new BlockedResult(false, null);
+
         // Check for allowed ip addresses (i.e. only one country is allowed to visit the site)
         // Always allow access from private IP addresses (those include local IP addresses)
-        if (!allowedIps.isEmpty() && !isPrivateIp(ip)) {
-            boolean ipAllowed = false;
-            for (IPListEntry entry : allowedIps) {
-                if (entry.ipList.matches(ip)) {
-                    ipAllowed = true; // We allow IP addresses as long as they match with one of the lists.
-                    break;
-                }
-            }
-            if (!ipAllowed) {
-                return new BlockedResult(true, "not in allowlist");
-            }
+        if (!isPrivateIp(ip) && !firewallLists.matchesAllowedIps(ip)) {
+            blockedResult = new BlockedResult(true, "not in allowlist");
```

Comment:
Why do we overwrite instead of returning?

e.g. if we know it's not in the allow list, we can already block the request?

Reply:
was meant to allow further IP matching (i.e. monitoring), will remove